components/text.h

```diff
@@ -15,26 +16,25 @@ namespace pong
 class Text : public UIComponent, public Component<Text>
 {
 public:
-    Text();
-    Text(const std::string& text, const std::string& fontName, float scale);
+    Text() = default;
+    Text(const std::string& text, Font* font, float scale);
 
     std::vector<OffsetGraphic> GetRenderables() override;
     BaseComponent* GetBaseComponent() override;
 
     std::string GetText() const;
     void SetText(const std::string& text);
-    void SetFont(const std::string& fontName);
+    void SetFont(Font* font);
     void SetScale(float scale);
     void SetColor(GLRGBAColor color);
     void RecomputeText();
 
     std::string mText {};
-    std::string mFont {};
+    Font* mFont { nullptr };
```

Comment:
I think you could get away with using a reference here and avoid the case where something is nullptr.